File: data/readable_format/cpp/file2255.cpp
#include <iostream>
#include <vector>
#include <cstdlib>

void sub(int N, std::vector<std::vector<double>>& u) {
    u[-N + N][N + N] = 0.0;
}

int main() {
    const int N = 10;
    std::vector<std::vector<double>> a(2 * N + 1, std::vector<double>(2 * N + 1, 1.0));

    sub(N, a);

    if (a[0][2 * N] != 0.0) {
        std::abort();
    }

    return 0;
}

File: data/readable_format/fortran/file2255.f90
subroutine sub(N,u)
      integer N
      double precision u(-N:N,-N:N)

      u(-N,N)=0d0
      return
end

program bug
      integer N
      double precision a(-10:10,-10:10)
      data a/441*1d0/
      N=10
      call sub(N,a)
      if (a(-N,N) .ne. 0d0) then
         print *, 'Test failed: a(-N,N) is not 0d0'
         call abort
      else
         print *, 'Test passed: a(-N,N) is 0d0'
      end if
end program bug